public/javascripts/SVLabel/src/SVLabel/mission/MissionProgress.js

```diff
@@ -125,6 +125,9 @@ function MissionProgress (svl, gameEffectModel, missionModel, modalModel, neighb
         neighborhoodContainer.setCurrentNeighborhood(neighborhood);
         neighborhoodModel.moveToANewRegion(neighborhoodId);
 
+        var currentTask = taskContainer.getCurrentTask();
```

Comment:
@Soben713 I think these two lines (128 & 129) fix the problem by ending the currentTask by calling `TaskContainer.endTask` method. 

Although I couldn't figure out why the issue was caused and the task was not marked `completed` after actually completing the task, I suspect it was because `TaskContainer.setCurrentTask` (line 135) in this method (`_updateTheCurrentNeighborhood`) was called *before* marking the current task as `completed`, which prevents finishing the other part of the code to actually end the current task. So explicitly ending the task here should fix the issue.